Classes/System/Configuration/TypoScriptConfiguration.php

```diff
@@ -359,6 +359,32 @@ public function getIndexQueueAllowedPageTypesArrayByConfigurationName(string $co
         return GeneralUtility::trimExplode(',', $result);
     }
 
+    /**
+     * Returns an array of all allowedPageTypes.
+     *
+     * plugin.tx_solr.index.queue.pages.allowedPageTypes
+     *
+     * @return array
+     */
+    public function getAllIndexQueueAllowedPageTypesArray(): array
+    {
+        $configuration = $this->configurationAccess->get('plugin.tx_solr.index.queue.');
+
+        if (!is_array($configuration)) {
+            return [];
+        }
+
+        $allowedPageTypes = [];
+        foreach ($configuration as $queueConfiguration) {
+            if (is_array($queueConfiguration)
+                && !empty($queueConfiguration['allowedPageTypes'])) {
+                $allowedPageTypes = array_merge($allowedPageTypes, GeneralUtility::trimExplode(',', $queueConfiguration['allowedPageTypes'], true));
+            }
+        }
```

Comment:
Is maybe the condition member for checking if the queue/type is enabled?
Also for example for pages = 1|0 setting: 

https://github.com/TYPO3-Solr/ext-solr/blob/cb5b5284a7cb55490e36477b0e4b3f3e59145616/Configuration/TypoScript/Solr/setup.typoscript#L24-L31

```suggestion
        foreach ($configuration as $queueName => $queueConfiguration) {
            if (is_array($queueConfiguration)
                && !empty($queueConfiguration['allowedPageTypes'])
                && $this->getIndexQueueConfigurationIsEnabled(rtrim($queueName,'.'))
            ) {
                $allowedPageTypes = array_merge($allowedPageTypes, GeneralUtility::trimExplode(',', $queueConfiguration['allowedPageTypes'], true));
            }
        }
```